tests/integration_tests/tests/agentless_tests/test_resume.py

```diff
@@ -0,0 +1,116 @@
+########
+# Copyright (c) 2019 Cloudify Platform Ltd. All rights reserved
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#        http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+#    * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+#    * See the License for the specific language governing permissions and
+#    * limitations under the License.
+
+import time
+from integration_tests import AgentlessTestCase
+from integration_tests.tests.utils import get_resource as resource
+
+
+class TestResumeMgmtworker(AgentlessTestCase):
+    wait_message = 'WAITING FOR FILE'
+    target_file = '/tmp/continue_test'
+
+    def _create_deployment(self):
+        dsl_path = resource("dsl/resumable_mgmtworker.yaml")
+        return self.deploy(dsl_path)
+
+    def _start_execution(self, deployment, operation):
+        return self.execute_workflow(
+            workflow_name='execute_operation',
+            wait_for_execution=False,
+            deployment_id=deployment.id,
+            parameters={'operation': operation,
+                        'run_by_dependency_order': True,
+                        'operation_kwargs': {
+                            'wait_message': self.wait_message,
+                            'target_file': self.target_file
+                        }})
+
+    def _wait_for_log(self, execution):
+        self.logger.info('Waiting for operation to start')
+        while True:
+            logs = self.client.events.list(
+                execution_id=execution.id, include_logs=True)
+            if any(self.wait_message == log['message'] for log in logs):
+                break
+            time.sleep(1)
+
+    def _restart_mgmtworker(self):
```

Comment:
This should probably be renamed to indicate why it's placing the target_file; I can understand why having read the rest of this code in context, but it'll likely be less clear in future.